Создано api и интегрирован ajax в сайт.

spectator196 opened the pull request:
build: создано api
refactor: переработан процесс получения информации для страниц посредством ajax
refactor: для красивого отображения страниц подключен BS

spectator196 committed:
Создана директория для решения задач.

spectator196 committed:
Решены 2 задачи по работе с классами.

nuberry commented on Class_tasks/class_task_2.php:
```diff
@@ -0,0 +1,26 @@
+
+Class SquareChecker{
+    
+    public $part_list;
```
Используй единую нотация - `camelCase`. В другом классе `camelCase`, в этом `snake_case`

nuberry commented on Class_tasks/class_task_2.php:
```diff
@@ -0,0 +1,26 @@
+    }
+}
+
+$figure_1= new SquareChecker;
+$figure_1->part_list=[20,20,20,20];
+print $figure_1->isSquare();// Квадрат
```
Можно побольше разных примеров для теста работоспособности.

nuberry commented on Class_tasks/class_task_1.php:
```diff
@@ -0,0 +1,27 @@
+
+}
+
+$animal_1= new Animal;
+$animal_1->type='Кошка';
+$animal_1->feedObj='Рыба';
+
+$animal_2= new Animal;
+$animal_2->type='Зебра';
+
+$animal_1->showMyAnimal();// Животное: Кошка Лакомство: Рыба
+$animal_2->showMyAnimal();// Дикое животное
```
Получится еще добавить тестовых кейсов?

nuberry commented on Class_tasks/class_task_2.php:
```diff
@@ -0,0 +1,26 @@
+        if (!isset($this->part_list)) {return 'Не указан массив со сторонами';}
+        if (count($this->part_list)!=4) {return 'Не корректное кол-во элементов фигуры';}
+
+        foreach ($this->part_list as $part){
+            if (!is_numeric($part)) { return 'Нужно вводить только числа';}
+            if ($part<0) {return 'Отрицательные цисла неприемлемы';}
+        }
+        
+        if (!(array_sum($this->part_list)/4==$this->part_list[0])){return 'Стороны квадрата должны быть равными';}
```
Неплохо. Но думаю можно было про равность сторон вынести в цикл, а не гонять через `array_sum`

nuberry reviewed (commented):
Хорошо. Также можешь попробовать добавить типизацию методам и свойствам

spectator196 committed:
Внесение правок в задачи:

- Все приведено к snake_case.
- Добавлена проверка на факт наличия свойства $type в show_my_animal
- Переработана проверка на совпадение всех елементов $part_list
- Добавлены дополнительные проверки работы классов.

spectator196 committed:
Обе задачи/класса приведены к camelCase

spectator196 committed:
Изменена структура примеров

spectator196 committed:
Создан api.php для обработки поступающих запросов

nuberry commented on api.php:
```diff
@@ -0,0 +1,58 @@
+} elseif ($_GET['path']=='applications')
+{
+    print(json_encode(get_db_full_table_applications([])));
+}
```
Зачем такое ветвление?
Не особо хорошо дважды проверять path на одно и тоже значение.

nuberry commented on api.php:
```diff
@@ -0,0 +1,58 @@
+
+if ($_GET['path']=='managers' and (isset($_GET['manager_id'])))
+{
+    print(json_encode(get_db_specific_managers(($_GET['manager_id']))));
```
Зачем у `print` скобки? 
почитай в документации про скобки у `print`  https://www.php.net/manual/ru/function.print.php и убери их использование

nuberry commented on tests.php:
```diff
@@ -0,0 +1,18 @@
+  $.get('api.php',{path: "managers"}, function(data) {
+    $('#result').append('Запись: '+data+'<br><br>');
+  });
+});
```
Вот этот весь код и предполагалось вынести в `script.js`.
В любом случае файла `tests.php` не должно быть.

Вероятнее всего, для упрощения, у тебя js скрипт будет внутри `<script>`, в каждом файле (add/edit/index php), вместо единого `script.js` так как допуская возникновения конфликтов селекторов и лучше пусть отдельно, пока, будет. Обновил в гугл документе

nuberry commented on api.php:
```diff
@@ -0,0 +1,58 @@
+if ($_POST['path']=='managers/update' and (isset($_POST['manager_data'])))
+{   
+    edit_db_application_data($_POST['manager_data']);
+}
```
Для информации: если научится корректно (см комментарии выше) обрабатывать метод запроса (PUT/DELETE/POST), то из `path` можно будет убрать `delete` и `update`.

nuberry replied:
А почему бы для обновления не взять за основу гет запрос с использованием id (например manager_id) заодно можно будет проверить наличие id.

Естественно придется переписать метод по работе с БД, т.к. ID будет отдельным аргументом.
У тебя данные для обновления приходят с формы, а id получаешь отдельно

spectator196 committed:
Полностью переработан api.php
Добавлен ajax в следующие роуты:
/managers/edit.php?manager_id=% (частично)
/managers/add.php
/managers
/applications/add.php
/applications

spectator196 committed:
Переработан api.php
Добавлена поддержка ajax в часть роутов

spectator196 committed:
Внесены правки в api.php и get_db_info.php
Внедрен Bootstrap
Внедрены функции апи для получения и отправки запросов

spectator196 committed:
Остальные доработки

spectator196 committed:
Добавлена ajax поддержка запросов на удаление
Интегрирован BS для стилизации сайта

nuberry commented on api.php:
```diff
@@ -0,0 +1,175 @@
+                    }
+                    break;
+                default:
+                    $result=('Некорректное обращение методом POST к api!');
```
1. Зачем тут скобочки. Также в других местах кода наблюдаю у `result` скобочки. Удали, обсуждали, что они не нужны.
2. Обсуждали, что можно сделать через константу массива с доступными методами.

nuberry commented on api.php:
```diff
@@ -0,0 +1,175 @@
+                    }
+                    break;
+                case 'managers':
+                    print_r($_POST);
+                    die();
```
Тестовый код. Удалить!

nuberry commented on applications/edit.php:
```diff
@@ -18,96 +18,83 @@
+                method: 'get',
+                dataType: 'json',
+                success: function (data) {
+                    console.log(data);
```
Тестовый код. Раз у тебя по факту запроса, что то выводится на экран, то вывод в консоль считается дебагом и не желательно его наличие в коде

nuberry commented on applications/index.php:
```diff
@@ -3,55 +3,77 @@
       <p><b>ID: </b><input type="text" name='id' size="20"></p>
       <p><b>ФИО: </b><input type="text" name='full_name' size="20"></p>
-      <button>Фильтровать</button>
+      <button class="btn btn-accept">Фильтровать</button>
```
accept, появился новый класс?

Замени на базовый

![image](https://user-images.githubusercontent.com/44833731/235649657-30d48989-e241-4ffb-af2d-9ffff99c55ed.png)

nuberry commented on applications/index.php:
```diff
@@ -3,55 +3,77 @@
-      <th>Действия</th>
-    </tr>
-    <?php foreach ($data_list as $record): ?>
+  <div class="container">
+    <table class="table table-bordered table-striped float-right" id='result_table'>
```
`float-right` не в тему. Удали

nuberry replied:
Неплохо было бы обернуть в 
```
<div class="row"><div class="col">TABLE</div></div>
```

nuberry commented on get_db_info.php:
```diff
@@ -173,17 +172,16 @@ function add_db_manager_data($new_data): void
 }
 
-function edit_db_application_data($update_params): void
+function edit_db_application_data($id,$update_params): void
```
Форматирование

nuberry commented on get_db_info.php:
```diff
@@ -210,6 +208,6 @@ function add_db_application_data($new_data): void
 function delete_db_application_data($data): void
 {
-    $sql_request = 'DELETE FROM applications WHERE id =' . $data['id'];
+    $sql_request = 'DELETE FROM applications WHERE id =' . $data;
```
`$data` - это не массив с ID?
Тогда логичнее аргумент переименовать в `$id`

nuberry commented on managers/add.php:
```diff
@@ -16,13 +16,34 @@
+                    data: form.serialize(),
+                    dataType: 'json',
+                    success: function (data) {
+                        console.log(data);
```
А к чему у тебя вообще этот запрос?

spectator196 replied:
Мы с вами в "прямом эфире" реализовывали 2 способа отправки запроса на добавление записи.

nuberry commented on managers/edit.php:
```diff
@@ -18,55 +14,78 @@
+                });
+            });
+        }); 
+    </script>
+
+    <script>
+        $(function () {
```
обсуждали про то что не обязательно несколько `<script>`. А несколько `$(function () {` вообще тут не к чему.

nuberry commented on managers/edit.php:
```diff
@@ -18,55 +14,78 @@
+
+    <script>
+        $(function () {
+            var manager_id = $('h1').attr("data-manager_id")
```
`data-*` лучше цеплять к форме или иному тэгу, а заголовок не особо подходящее место для этого.
Но можно итак оставить

nuberry commented on managers/index.php:
```diff
@@ -13,14 +13,55 @@
+        method: 'get',
+        dataType: 'json',
+        success: function (data) {
+          console.log(data);
```
Аналогично

nuberry commented on managers/index.php:
```diff
@@ -31,27 +72,14 @@
   </div>
 
-  <table border="1">
+  <table class="table table-bordered table-striped float-right" id='result_table'>
```
1. нет оберток
2. float-right не в тему

nuberry commented on tests.php:
```diff
@@ -0,0 +1,55 @@
+<html>
```
Не особо уверен, что этот файл нужен.
Он тестовый и не к чему тут

spectator196 committed:
Внесены правки в соответствии с комментариями

spectator196 committed:
Добавлена улучшенная обработка ошибок api

spectator196 committed:
Заменен массив ошибок на отдельные константы
